src/Context&functions/paramContext.jsx

```diff
@@ -0,0 +1,22 @@
+import { createContext, useEffect, useState } from "react";
+
+const ParamContext = createContext();
+const ParamProvider = ({ products, productId, children }) => {
+  const [paramItem, setParamItem] = useState(productId);
+
+  useEffect(() => {
+    const found = products.find((p) => p.id === productId);
```

Comment:
```suggestion
    const product = products.find((p) => p.id === productId);
```
Variables must have a context